### PR TITLE
Y2K-1805 Role Picker: Add internal selection management

### PR DESCRIPTION
**Pull request by Daniel-Wang**
Changes to RolePicker
* Refactored redundantly maintained `role selection state` to be managed within the component
* Added a `count` prop for pluralization

### CODE REVIEW

File: shared/teams/role-picker/index.tsx [hunk 1]
```diff
@@ -280,7 +279,10 @@ const Header = () => (
 )
 
 const RolePicker = <IncludeSetIndividually extends boolean>(props: Props<IncludeSetIndividually>) => {
-  const selectedRole = filterRole(props.selectedRole || props.presetRole)
+  const filteredRole = filterRole(props.presetRole)
+  const [selectedRole, setSelectedRole] = React.useState<Role<IncludeSetIndividually>>(
```
Comment (jakob223) [inline shared/teams/role-picker/index.tsx:283]: if presetRole changes due to the parent, we should probably change selectedRole. Maybe a useEffect, or maybe include it in the onOpen

**Comment by Daniel-Wang** [comment]
For each refactored picker, I have tried to locate and test it on mobile and desktop. But I might have missed something from the volume of refactored pickers.

File: shared/team-building/role-picker-header-action.tsx [hunk 2]
```diff
@@ -15,18 +14,18 @@ const RolePickerHeaderAction = (props: Props) => {
...
   return (
     <FloatingRolePicker
+      presetRole={props.rolePickerProps.selectedRole}
       open={rolePickerOpen}
       onConfirm={onConfirm}
-      onSelectRole={props.rolePickerProps.onSelectRole}
-      selectedRole={props.rolePickerProps.selectedRole}
       onCancel={() => setRolePickerOpen(false)}
-      confirmLabel={`Add as ${pluralize(props.rolePickerProps.selectedRole, props.count)}`}
+      confirmLabel={`Add as`}
```
Comment (jakob223) [inline shared/team-building/role-picker-header-action.tsx:28]: hmm this no longer pluralizes, which is unfortunate

Comment (jakob223) [reply shared/team-building/role-picker-header-action.tsx:28]: Looks like you added a prop for that, wanna include here

File: shared/teams/add-members-wizard/confirm.tsx [hunk 3]
```diff
@@ -218,12 +217,11 @@ const RoleSelector = ({disabledRoles, memberCount}: RoleSelectorProps) => {
         open={showingMenu}
         presetRole={storeRole}
         onCancel={storeRole === role ? () => setShowingMenu(false) : undefined}
-        selectedRole={role || 'writer'}
-        onSelectRole={onSelectRole}
         onConfirm={onConfirmRole}
-        confirmLabel="Save"
+        confirmLabel="Save as"
```
Comment (jakob223) [inline shared/teams/add-members-wizard/confirm.tsx:221]: this is awkward. Maybe take either a string e.g. `'save'` or a function e.g. ```role => `Let in as ${pluralize("role", count)}` ``` and then get rid of count param?

File: shared/teams/common/selection-popup.tsx [hunk 4]
```diff
@@ -309,15 +305,13 @@ const EditRoleButton = ({members, teamID}: {teamID: Types.TeamID; members: strin
...
-      selectedRole={role}
-      onSelectRole={setRole}
       onConfirm={onChangeRoles}
       onCancel={() => setShowingPicker(false)}
       position="top center"
       open={showingPicker}
       disabledRoles={disabledReasons}
       waiting={waiting}
-      confirmLabel={role ? `Make ${pluralize(role, members.length)}` : undefined}
+      count={members.length}
```
Comment (jakob223) [inline shared/teams/common/selection-popup.tsx:314]: unused?

File: shared/teams/role-picker/index.tsx [hunk 5]
```diff
@@ -280,7 +291,15 @@ const Header = () => (
...
+    filteredRole ?? ('reader' as Role<IncludeSetIndividually>)
+  )
+  // React.useEffect(() => {
```
Comment (jakob223) [inline shared/teams/role-picker/index.tsx:298]: commented out?

File: shared/teams/team/member/index.new.tsx [hunk 6]
```diff
@@ -367,14 +366,13 @@ const NodeNotInRow = (props: NodeNotInRowProps) => {
         {props.node.canAdminister && (
           <Kb.Box2 direction="horizontal" alignSelf="center">
             <FloatingRolePicker
-              onConfirm={() => {
+              presetRole={'writer'}
```
Comment (jakob223) [inline shared/teams/team/member/index.new.tsx:369]:
```suggestion
              presetRole="writer"
```

**Comment by Daniel-Wang** [comment]
Decided to take out the confirmLabel and default to `Save`, seems most consistent with design and also seems to be okay with all situations.

**Pull request merged**